src/alpaka/Framework/WaitingTaskWithArenaHolder.cc

```diff
@@ -21,9 +21,23 @@ namespace edm {
   // Note that the arena will be the one containing the thread
   // that runs this constructor. This is the arena where you
   // eventually intend for the task to be spawned.
-  WaitingTaskWithArenaHolder::WaitingTaskWithArenaHolder(WaitingTask* iTask)
-      : m_task(iTask), m_arena(std::make_shared<tbb::task_arena>(tbb::task_arena::attach())) {
+  WaitingTaskWithArenaHolder::WaitingTaskWithArenaHolder(tbb::task_group& iGroup, WaitingTask* iTask)
+      : m_task(iTask), m_group(&iGroup), m_arena(std::make_shared<tbb::task_arena>(tbb::task_arena::attach())) {
     m_task->increment_ref_count();
+    m_handle = std::make_shared<tbb::task_handle>(m_group->defer([task = m_task]() {
```

Comment:
I'm not terribly happy with the approach I took for `tbb::task_handle` here, e.g. there are now three members doing reference counting (two `shared_ptr`s implicitly and `m_task` explicitly), but I'd leave improvements (likely a redesign) to a later time.